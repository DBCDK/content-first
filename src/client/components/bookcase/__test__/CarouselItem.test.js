import React from 'react';
import renderer from 'react-test-renderer';

import CarouselItem from '../CarouselItem.component';

jest.mock(
  '../../general/CheckmarkConnected.component',
  () => 'CheckmarkConnected'
);
jest.mock('../../general/Link.component', () => 'Link');

const description = 'lorem ipsum . . .';

const book = {
  pid: '870970-basis:51642899',
  titel: 'Alt det lys vi ikke ser',
  creator: 'Anthony Doerr',
  description: 'Roman nder 2. verdenskrig'
};

describe('CarouselItem', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <CarouselItem
          description={description}
          onClick={() => {
            this.nextBook('next');
          }}
          book={book}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
