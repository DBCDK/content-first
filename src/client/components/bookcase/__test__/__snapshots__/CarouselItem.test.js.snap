// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem renders initial component 1`] = `
<div
  className="carousel-container "
>
  <div
    className="carousel"
  >
    <div
      className="col-xs-4 carousel-img"
    >
      <Link
        href="/værk/870970-basis:51642899"
      >
        <img
          alt=""
          className="high-contrast "
          onLoad={undefined}
          src="/default-book-cover.png"
          style={undefined}
        />
      </Link>
    </div>
    <div
      className="col-xs-8 text-left carousel-text"
    >
      <div
        className="col-xs-12 carousel-title"
      >
        <h1 />
      </div>
      <div
        className="col-xs-12 carousel-creator"
      >
        <Link
          href="/værk/870970-basis:51642899"
        >
          <h2>
            Anthony Doerr
          </h2>
        </Link>
      </div>
      <div
        className="col-xs-12 carousel-description"
      >
        <react-truncate-markup
          lines={6}
        >
          <p>
            lorem ipsum . . .
          </p>
        </react-truncate-markup>
      </div>
      <div
        className="col-xs-12"
      >
        <CheckmarkConnected
          book={
            Object {
              "book": Object {
                "creator": "Anthony Doerr",
                "description": "Roman nder 2. verdenskrig",
                "pid": "870970-basis:51642899",
                "titel": "Alt det lys vi ikke ser",
              },
            }
          }
          origin="Fra bogreol"
        />
      </div>
    </div>
    <div
      className="clear"
    />
  </div>
</div>
`;
